src/utils/apiRequests/api.ts

```diff
@@ -251,24 +252,29 @@ export async function putAuthenticatedRequest(
   errorHandler?: Function
 ): Promise<any> {
   if (validateToken(token)) {
-    const res = await fetch(process.env.API_ENDPOINT + url, {
-      method: 'PUT',
-      body: JSON.stringify(data),
-      headers: {
-        'Content-Type': 'application/json',
-        'tenant-key': `${TENANT_ID}`,
-        'X-SESSION-ID': await getsessionId(),
-        Authorization: `Bearer ${token}`,
-        'x-locale': `${
-          localStorage.getItem('language')
-            ? localStorage.getItem('language')
-            : 'en'
-        }`,
-      },
-    });
-    const result = await res.json();
-    handleApiError(res.status, errorHandler);
-    return result;
+    try{
+
+      const res = await fetch(process.env.API_ENDPOINT + url, {
+        method: 'PUT',
+        body: JSON.stringify(data),
+        headers: {
+          'Content-Type': 'application/json',
+          'tenant-key': `${TENANT_ID}`,
+          'X-SESSION-ID': await getsessionId(),
+          Authorization: `Bearer ${token}`,
+          'x-locale': `${
+            localStorage.getItem('language')
+              ? localStorage.getItem('language')
+              : 'en'
+          }`,
+        },
+      });
+      const result = await res.json();
+      handleApiError(res.status, errorHandler);
+      return result;
+    }catch(err){
+        throw new Error('500')
```

Comment:
@sunilsabatp Need to handle this in a better way, since not all the errors will be 500. Throwing error like this will make us unable to get more info about the error.